src/interface/viewport.c

```diff
@@ -864,29 +864,179 @@ void sub_688485(){
  * @param bound_box_offset_y (0x009DEA54)
  * @param bound_box_offset_z (0x009DEA56)
  * @param rotation (ebp)
- * @return ??
+ * @return (inverse CF)
  */
-int sub_98199C(
+bool sub_98199C(
 	int image_id,
 	sint8 x_offset, sint8 y_offset,
 	sint16 bound_box_length_x, sint16 bound_box_length_y, sint8 bound_box_length_z,
 	int z_offset,
 	uint16 bound_box_offset_x, uint16 bound_box_offset_y, uint16 bound_box_offset_z,
 	uint32 rotation
 ) {
-	RCT2_GLOBAL(RCT2_ADDRESS_PAINT_BOUNDBOX_OFFSET_X, uint16) = bound_box_offset_x;
-	RCT2_GLOBAL(RCT2_ADDRESS_PAINT_BOUNDBOX_OFFSET_Y, uint16) = bound_box_offset_y;
-	RCT2_GLOBAL(RCT2_ADDRESS_PAINT_BOUNDBOX_OFFSET_Z, uint16) = bound_box_offset_z;
+	if (RCT2_GLOBAL(0xF1AD28, uint32) == 0) {
+		return sub_98197C(
+			image_id,
+			x_offset, y_offset,
+			bound_box_length_x, bound_box_length_y, bound_box_length_z,
+			z_offset,
+			bound_box_offset_x, bound_box_offset_y, bound_box_offset_z,
+			rotation
+		) == 1;
+	}
+
+	//Not a paint struct but something similar
+	paint_struct *ps = RCT2_GLOBAL(0xEE7888, paint_struct*);
+	if ((uint32) ps >= RCT2_GLOBAL(0xEE7880, uint32)) {
+		return false;
+	}
+
+	ps->image_id = image_id;
+
+	uint32 image_element = image_id & 0x7FFFF;
+	rct_g1_element *g1Element = gfx_get_g1_element(image_element);
+
+	rct_xyz16 coord_3d = {
+		.x = x_offset, // ax
+		.y = y_offset, // cx
+		.z = z_offset
+	};
+
+	switch (rotation) {
+		case 1:
+			coord_3d.x = -y_offset;
+			coord_3d.y = x_offset;
+			break;
+
+		case 2:
+			coord_3d.x = -x_offset;
+			coord_3d.y = -y_offset;
+			break;
+
+		case 3:
+			coord_3d.x = x_offset;
+			coord_3d.x = -y_offset;
```

Comment:
You're changing `coord_3d.x` twice here. Didn't you mean to use `coord_3d.x` on one of the occasions?